samples/sensor/stm32_temp_sensor/src/main.c

```diff
@@ -9,11 +9,19 @@
 #include <drivers/sensor.h>
 #include <sys/printk.h>
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_temp)
+#define TEMP_NODE DT_INST(0, st_stm32_temp)
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_temp_cal)
+#define TEMP_NODE DT_INST(0, st_stm32_temp_cal)
+#else
+#define TEMP_NODE DT_INVALID_NODE
```

Comment:
Let's use an `#error ` directive here to directly detect configuration issues at build time